05-js-basico/mariano-queirel/calculator.js

```diff
@@ -0,0 +1,68 @@
+const calculator = document.querySelector('.calculator');
+const keys = calculator.querySelector('.keys');
+const display = document.querySelector('.display');
+display.innerHTML = 0;
+
+keys.addEventListener('click', e => {
+  if (e.target.matches('button')) {
+    const key = e.target;
+    const action = key.dataset.action;
+    const keyContent = key.textContent;
+    const displayedNumber = display.textContent;
+    const previousKeyType = calculator.dataset.previousKeyType;
+
+
+    if (action != null) {
+      if (action === 'decimal') {
+        display.textContent = displayedNumber + keyContent;
+      }
+
+      if (action === 'clear') {
+        display.textContent = 0;
+      }
+
+      if (action === 'calculate') {
+        const firstValue = calculator.dataset.firstValue;
+        const operator = calculator.dataset.operator;
+        const secondValue = displayedNumber;
+
+        if (firstValue != null && secondValue != null) {
+          display.textContent = calculate(firstValue, operator, secondValue);
+        }
+      }
+
+      if (action === 'sum' || action === 'subtract' || action === 'multiply' || action === 'divide') {
+        calculator.dataset.previousKeyType = 'operator';
+        calculator.dataset.firstValue = display.textContent;
+        calculator.dataset.operator = action;
+      }
+    }
+
+    if (action == null) {
+      calculator.dataset.previousKeyType = 'number';
+      if (displayedNumber == 0 || previousKeyType === 'operator' ) {
+        display.textContent = keyContent;
+      } else {
+        display.textContent = displayedNumber + keyContent;
+      }
+
+    }
+  }
+})
+
+function calculate(firstValue, operator, secondValue) {
```

Comment:
Agregar comentario a método de la función. De la siguiente manera:

    /*
    * Method: calculate
    * Description: Method responsible to do something
    * Params: {String} firstValue
    * Params: {String} operator
    * Return: Integer
    * Author: Mariano
    **/